applications/nrf_desktop/src/modules/usb_state_pm.c

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#include <zephyr.h>
+#include <zephyr/types.h>
+
+#include "usb_event.h"
+
+#define MODULE usb_state_pm
+#include <caf/events/module_state_event.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_DESKTOP_USB_STATE_LOG_LEVEL);
+
+#include <caf/events/power_manager_event.h>
+#include <caf/events/force_power_down_event.h>
+
+
+#define SUSPEND_DELAY K_MSEC(1000)
+
+static struct k_work_delayable suspend_trigger;
+
+
+static void suspend_worker(struct k_work *work)
+{
+	LOG_INF("Suspending");
+	force_power_down();
+}
+
+static bool event_handler(const struct event_header *eh)
+{
+	if (is_usb_state_event(eh)) {
+		const struct usb_state_event *event = cast_usb_state_event(eh);
+
+		LOG_INF("USB state change detected");
+		k_work_cancel_delayable(&suspend_trigger);
+
+		switch (event->state) {
+		case USB_STATE_POWERED:
+		case USB_STATE_ACTIVE:
+			power_manager_restrict(MODULE_IDX(MODULE), POWER_MANAGER_LEVEL_ALIVE);
+			return false;
```

Comment:
If there will be something to do, these returns should be switched to breaks and one return left after switch case.

Reply:
fixed